src/d3a/models/state.py

```diff
@@ -115,6 +128,37 @@ def __init__(self,
         self._battery_energy_per_slot = 0.0
         self._used_storage_share = [EnergyOrigin(initial_energy_origin, initial_capacity_kWh)]
 
+    def get_state(self):
+        return {
+            "pledged_sell_kWh": convert_pendulum_to_str_in_dict(self.pledged_sell_kWh),
+            "offered_sell_kWh": convert_pendulum_to_str_in_dict(self.offered_sell_kWh),
+            "pledged_buy_kWh": convert_pendulum_to_str_in_dict(self.pledged_buy_kWh),
+            "offered_buy_kWh": convert_pendulum_to_str_in_dict(self.offered_buy_kWh),
+            "charge_history": convert_pendulum_to_str_in_dict(self.charge_history),
+            "charge_history_kWh": convert_pendulum_to_str_in_dict(self.charge_history_kWh),
+            "offered_history": convert_pendulum_to_str_in_dict(self.offered_history),
+            "used_history": convert_pendulum_to_str_in_dict(self.used_history),
+            "energy_to_buy_dict": convert_pendulum_to_str_in_dict(self.energy_to_buy_dict),
+            "energy_to_sell_dict": convert_pendulum_to_str_in_dict(self.energy_to_sell_dict),
+            "used_storage": self._used_storage,
+            "battery_energy_per_slot": self._battery_energy_per_slot,
+        }
+
+    def load_state(self, state_dict):
```

Comment:
Just FYI: I was confused at first. I thought that this is the state of the load. If you can think of another naming loading the state, please change. If not, its fine really.

Reply:
Maybe `restore_state` ?

Reply:
yes, great!